Add module frame

tobes opened the pull request:
The frame module.  This allow modules to be grouped.  This is useful if you wish to add more than one module to a 'slot' in a group.

```
group {
    frame {
        volume_status {}
        battery_level {}
    }

    time {}
}
```

ultrabug commented:
We ought to document this more than in the module itself since it's a great feature ! Would you care updating the documentation here https://github.com/ultrabug/py3status/tree/master/doc#group_modules with this feature please ?

tobes commented:
Sure

tobes commented:
I've added some documentation as requested.

I also uncovered a bug, that I fixed in this PR.  Some modules need to have the `output_modules` dict in core.py available to get content from other modules, but were running before it was built.  We now build the dict before starting the modules.

tobes commented:
I'm thinking of adding an open/close button to frame like the group module has as I think this is a useful space saving feature.

Would that be acceptable?  Alternatively that could be a separate module but it feels that pushing it into frame would be more sensible.   Then any module you wanted to be open/close-able you could just put in a frame.

ultrabug commented:
> I'm thinking of adding an open/close button to frame like the group module has as I think this is a useful space saving feature.

Sure

tobes commented:
Ok, this now has the button added and a way to change the separators between the modules.

Anyhow I think I'm finished on this now.

I also added `py3.is_my_event(event)` helper that is useful for containers like group, hope that is ok.

ultrabug commented on py3status/modules/frame.py:
```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+"""
+Group modules and treat as a single one.
```
"and display them as a single one" instead ?

ultrabug commented:
minor docstring remark and I merge :+1: thanks a lot !

tobes commented:
typo fixed

ultrabug commented:
awesome :D
